hack/build-and-push.sh

```diff
@@ -431,6 +436,34 @@ attach_migration_file() {
     return 0
 }
 
+# Find previous bundle that has a migration and output its digest.
+find_previous_migration_bundle_digest() {
+    local -r task_name=${1:?Missing task name}
+    local -r task_version=${2:?Missing task version}
+    local -r repo=${TEST_REPO_NAME:-task-${task_name}}
+
+    local -r url_list_recent_tags="https://quay.io/api/v1/repository/${QUAY_NAMESPACE}/${repo}/tag/?onlyActiveTags=true&limit=30"
+    local -r expr_filter_tags=".tags[] | select(.name | test(\"^(${task_name}-)?${task_version}-[0-9a-f]+\$\")) | .manifest_digest"
```

Comment:
I think we can check if we push into a test location (all bundles into the same image repo) and if not, just get only 1 tag without filtering by name.
Also comments would be appreciated. Thanks.